test/client/http_service.test.js

```diff
@@ -0,0 +1,265 @@
+
+      const httpService = await th
+        .app()
+        .services()
+        .create({ type: SERVICE_TYPE, name: SERVICE_NAME });
+      await th
+        .app()
+        .services()
+        .service(httpService._id)
+        .rules()
+        .create({
+          name: 'rule1',
+          actions: ['get', 'post', 'put', 'patch', 'delete', 'head'],
+          when: {}
+        });
+      service = th.stitchClient.service(SERVICE_TYPE, SERVICE_NAME);
+    });
+
+    afterEach(async() => th.cleanup());
+
+    describe('Submitting a GET request', () => {
+      const requestHandler = (req, res) => {
+        res.writeHead(200, { 'Content-Type': 'text/plan' });
+        res.end('POW!');
+      };
+
+      it('with url should work', async() => {
+        handleRequest = requestHandler;
+
+        const { statusCode, body } = await service.get(TEST_SERVER_URL);
+
+        expect(statusCode).toBe(200);
+        expect(body.toString()).toBe('POW!');
+      });
+
+      it('with options should work', async() => {
+        handleRequest = requestHandler;
+
+        const { statusCode, body } = await service.get({ url: TEST_SERVER_URL });
+
+        expect(statusCode).toBe(200);
+        expect(body.toString()).toBe('POW!');
+      });
+    });
+
+    describe('Submitting a POST request', () => {
+      const requestHandler = async(req, res) => {
+        const body = await readRequestBody(req);
+        res.writeHead(201, { 'Content-Type': 'text/plan' });
+        res.end(body);
+      };
+
+      it('with url should work', async() => {
+        handleRequest = requestHandler;
+
+        const { statusCode, body } = await service.post(TEST_SERVER_URL);
+
+        expect(statusCode).toBe(201);
+        expect(body.toString()).toBe('');
+      });
+
+      it('with options should work (raw body)', async() => {
+        handleRequest = requestHandler;
+
+        const requestBody = { foo: 'bar' };
+        const { statusCode, body } = await service.post({
+          url: TEST_SERVER_URL,
+          body: requestBody,
+          encodeBodyAsJSON: true
+        });
+
+        expect(statusCode).toBe(201);
+        expect(body.toString()).toBe(JSON.stringify(requestBody));
+      });
+
+      it('with options should work (stringified body)', async() => {
+        handleRequest = requestHandler;
+
+        const requestBody = JSON.stringify({ foo: 'bar' });
+        const { statusCode, body } = await service.post({
+          url: TEST_SERVER_URL,
+          body: requestBody
+        });
+
+        expect(statusCode).toBe(201);
+        expect(body.toString()).toBe(requestBody);
+      });
+    });
+
+    describe('Submitting a PUT request', () => {
+      const requestHandler = async(req, res) => {
+        const body = await readRequestBody(req);
+        res.writeHead(204, { 'Content-Type': 'text/plan' });
+        res.end(body);
+      };
+
+      it('with url should work', async() => {
```

Comment:
I think we should make these test descriptions more precise, where possible. For instance, this one is specifically testing that you can invoke `.put` using a string argument instead of an object. So maybe a better description might be "with a string argument should send a successful request", etc.